pytorch_lightning/loops/dataloader/evaluation_loop.py

```diff
@@ -271,17 +271,86 @@ def _on_evaluation_epoch_end(self) -> None:
         self.trainer._call_lightning_module_hook("on_epoch_end")
         self.trainer.logger_connector.on_epoch_end()
 
+    def _get_keys(self, iterable: Union[List, dict]) -> Iterable[str]:
+        if isinstance(iterable, list):
+            for i in iterable:
+                yield from self._get_keys(i)
+        elif isinstance(iterable, dict):
+            for k, v in iterable.items():
+                if isinstance(v, dict):
+                    yield from self._get_keys(v)
+                else:
+                    yield k
+
+    def _find_value(self, data: dict, target: str) -> Iterable[Any]:
+        for k, v in data.items():
+            if k == target:
+                yield v
+            elif isinstance(v, dict):
+                yield from self._find_value(v, target)
+
     def _print_results(self, results: List[_OUT_DICT], stage: RunningStage) -> None:
-        # TODO: this could be updated to look nicer
-        from pprint import pprint
-
-        print("-" * 80)
-        for i, metrics_dict in enumerate(results):
-            print(f"DATALOADER:{i} {stage.upper()} RESULTS")
-            pprint(
-                {
-                    k: (v.item() if v.numel() == 1 else v.tolist()) if isinstance(v, torch.Tensor) else v
-                    for k, v in metrics_dict.items()
-                }
-            )
-            print("-" * 80)
+        if _RICH_AVAILABLE:
+            from rich.console import Console
+            from rich.table import Table
+
+            console = Console()
+            table = Table()
+
+            table.add_column("Metric", justify="center", style="cyan")
+
+            for i, result in enumerate(results):
+                clean_keys = [i.split("/dataloader_idx_")[0] for i in result.keys()]
+                results[i] = dict(zip(clean_keys, result.values()))
+
+            unique_keys: List[str] = sorted(list(set(self._get_keys(results))))
+            rows: List[List[str]] = [[i] for i in unique_keys]
+
+            for i, metrics in enumerate(results):
+                table.add_column(f"DATALOADER {i}", justify="center", style="magenta")
+
+                for metric in rows:
+                    v = list(self._find_value(metrics, metric[0]))
+                    if v:
+                        metric.append(f"{v[0]}")
+                    else:
+                        metric.append(" ")
+
+            for row in rows:
+                table.add_row(*row)
+
+            console.print("", table)
+
+        else:
+            import os
+
+            term_size = os.get_terminal_size()
```

Comment:
As a 4k screen user, this seems way too much, especially on pychar, where the terminal is anchored to the bottom.

I think something like max(120, actual values width) would be a better choice

120 because because our code is formatted to 120 columns max. Could also be 80 or 88

Reply:
I've added the max 120 in my latest commit to have a minimum width of 120 if this is what was intended?

Reply:
Sorry! This is what I meant: f78f4bb